src/pipeline.cc

```diff
@@ -777,6 +783,44 @@ class PipelineWorker : public AsyncWorker {
         }
       }
 
+      // Apply bitwise boolean operation between images
+      if (baton->booleanOp != VIPS_OPERATION_BOOLEAN_LAST &&
+          (baton->booleanBufferInLength > 0 || !baton->booleanFileIn.empty())) {
+        VImage booleanImage;
+        ImageType booleanImageType = ImageType::UNKNOWN;
+        if (baton->booleanBufferInLength > 0) {
```

Comment:
I feel like there's potential for extracting this `new_from_buffer`/`new_from_file` logic given it appears in a couple of places now. I'm happy to do this later though; this is more of a note-to-self.